languages/rust/exercises/concept/entry-api/.meta/example.rs

```diff
@@ -1,14 +1,14 @@
 use std::collections::HashMap;
 
 pub fn can_construct_note(magazine: &[&str], note: &[&str]) -> bool {
-    let mut counts = HashMap::with_capacity(magazine.len());
+    let mut counts: HashMap<&str, i32> = HashMap::with_capacity(magazine.len());
```

Comment:
Why the type annotation here? Did the inferencer fail?

Reply:
Yea, Rust Analyzer complains that it can't infer the type of the value.

Reply:
Hmm, Rust Analyzer is more limited than the actual compiler. I prefer to annotate only when the compiler can't figure out what's going on, even if the Analyzer can't. 

That's a personal preference though, and we haven't defined a type annotation policy, so it should not be held as binding.

Reply:
potential future maintenance issue: `rustc`'s type inference occasionally gets slightly stronger, so we can't actually know for certain what-all type annotations are superfluous without tediously removing them and testing.

Reply:
As it happens, as of v1.43.1, `rustc`'s type inference can't figure this out without the annotation anyway: https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=63ad5c03979dc9403bd8244d340ef3ae. 

It's clearly correct that the maintainers shouldn't assume the burden of periodically removing annotations as type inference improves; this can stand as it is.

Reply:
On a tangential note, this makes me wonder if it would be useful to have a feature in the inferencer, or perhaps in Rust Analyzer, that raises a notification when explicit type annotations match the inferred type to tell you that the annotation isn't needed 🤔